src/arith.h

```diff
@@ -435,24 +435,29 @@ bool solovay_strassen1(T a, T n)
 {
     const T exponent = (n - 1) / 2;
     const T res = exp_mod(a, exponent, n);
-    int j = jacobi(a, n);
+    SignedDoubleSizeVal<T> _a = a;
+    SignedDoubleSizeVal<T> _n = n;
+    int j = jacobi<T>(_a, _n);
```

Comment:
Why do you need `_a` and `_n`?

The input type of `jacoby` are already `SignedDoubleSizeVal<T>` so what was the problem?